sdk/confidentialledger/Azure.Security.CodeTransparency/src/CodeTransparencyClient.cs

```diff
@@ -101,68 +106,122 @@ bool CertValidationCheck(X509Certificate2 cert)
             return new HttpPipelineTransportOptions { ServerCertificateCustomValidationCallback = args => CertValidationCheck(args.Certificate) };
         }
 
-        /// <summary> Post an entry to be registered on the CodeTransparency instance. The returned operation will have the entry id when it is complete. </summary>
-        /// <param name="body"> A raw CoseSign1 signature. </param>
-        /// <param name="cancellationToken"> The cancellation token to use. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="body"/> is null. </exception>
-        public virtual Operation<GetOperationResult> CreateEntry(BinaryData body, CancellationToken cancellationToken = default)
+        /// <summary>
+        /// Verify the receipt integrity against the COSE_Sign1 envelope
+        /// and check if receipt was endorsed by the given service certificate.
+        /// In the case of receipts being embedded in the signature then verify
+        /// all of them.
+        /// Calls <!-- see cref="CcfReceiptVerifier.VerifyTransparentStatementReceipt(JsonWebKey, byte[], byte[])"/> for each receipt found in the transparent statement.-->
+        /// </summary>
+        /// <param name="transparentStatementCoseSign1Bytes">Receipt cbor or Cose_Sign1 (with an embedded receipt) bytes.</param>
+        /// <param name="signedStatement">The input signed statement in Cose_Sign1 cbor bytes.</param>
+        public void RunTransparentStatementVerification(byte[] transparentStatementCoseSign1Bytes, byte[] signedStatement)
```

Comment:
When testing locally I was a bit confused by the names of the parameters. My personal preference would be to have 2 separate methods: one for transparent statement and the other for signed statement+receipt. But maybe renaming parameters would help as well